### PR TITLE
Comments attachments order

### PR DESCRIPTION
**Pull request by colinyoung87**
- orders comments and attachments most recent first
- reverse order them in Angular
- include attachments in comments serializer

### CODE REVIEW

File: app/models/story.rb [hunk 1]
```diff
@@ -15,9 +15,10 @@ class Story < ActiveRecord::Base
   belongs_to :project, touch: true
   belongs_to :owner, class_name: "User"
   has_many :tasks, -> { order(completed: :asc, order: :asc, created_at: :asc) }
-  has_many :attachments
-  has_many :comments, as: :commentable
-  has_many :users, -> { uniq }, through: :comments, source: :owner
+  has_many :attachments, -> { order(created_at: :desc) }
+  has_many :comments, -> { order(created_at: :desc) }, as: :commentable
+  has_many :comments_unordered, as: :commentable, class_name: "Comment"
```
Comment (colinyoung87) [inline app/models/story.rb]: the `uniq` call below for `users` was blowing up when an order was passed through to the comments

File: app/serializers/comment_serializer.rb [hunk 2]
```diff
@@ -1,12 +1,18 @@
...
   include TagHelper
 
-  attributes :id, :content, :markdown, :commentable_type, :commentable_id, :owner_id, :created_at, :updated_at, :team_id, :owner
+  attributes :id, :content, :markdown, :commentable_type, :commentable_id, :owner_id, :created_at, :updated_at, :team_id, :owner, :attachments
 
   def owner
     ActiveModel::SerializableResource.new(object.owner).serializable_hash[:user]
   end
 
+  def attachments
```
Comment (colinyoung87) [inline app/serializers/comment_serializer.rb]: `has_many :attachments` wasn't working here, not sure why - this was my workaround to make sure the serializer was used

File: app/models/story.rb (hunk 1 above)
Comment (phawk) [inline app/models/story.rb]: Yeah, you can’t call order on a field that’s not in the select statement. Adding order to the associations can limit their uses in a join.

File: app/serializers/comment_serializer.rb (hunk 2 above)
Comment (phawk) [inline app/serializers/comment_serializer.rb]: Has happened me in a few places as well, tried to read the source code of AMS to figure out what situations it loads in the `has_many`s or when it doesn’t, I’ve still no clue. Kinda why I’d consider using that [jbuilder](https://github.com/rails/jbuilder) thing to build up JSON, even though it’s more verbose it’s much easier to understand what’s actually going on and you don’t have any headaches to figure out when the magic acts up.

**Pull request merged**